dapps/multisig-toolkit/src/components/preview-effects/EffectsPreview.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { DryRunTransactionBlockResponse } from '@mysten/sui.js/src/client';
+import * as Tabs from '@radix-ui/react-tabs';
+import { useState } from 'react';
+
+import { Textarea } from '../ui/textarea';
+import { BalanceChanges } from './partials/BalanceChanges';
+import { Events } from './partials/Events';
+import { ObjectChanges } from './partials/ObjectChanges';
+import { Overview } from './partials/Overview';
+import { Transactions } from './partials/Transactions';
+
+export function EffectsPreview({ output }: { output: DryRunTransactionBlockResponse }) {
+	const [tab, setTab] = useState('balance-changes');
+
+	const { objectChanges, balanceChanges } = output;
+
+	const tabs = [
+		{
+			name: 'balance-changes',
+			title: 'Balance Changes',
+			count: balanceChanges?.length,
+		},
+		{
+			name: 'object-changes',
+			title: 'Object Changes',
+			count: objectChanges?.length,
+		},
+		{
+			name: 'events',
+			title: 'Events',
+			count: output.events.length,
+		},
+		{
+			name: 'transactions',
+			title: 'Transactions',
+			count:
+				output.input.transaction.kind === 'ProgrammableTransaction'
+					? output.input.transaction.transactions.length
+					: 0,
+		},
+		{
+			name: 'json',
+			title: 'Raw JSON',
+		},
+	];
+
+	return (
+		<>
+			<Overview output={output} />
+			<Tabs.Root value={tab} onValueChange={setTab} className="w-full">
+				<Tabs.List className="flex overflow-x-auto border-b ">
+					{tabs.map((tab, index) => {
```

Comment:
Kind of odd to have the tabs defined in the list but the tab content defined explicitly below.